### DISCUSSION TITLE
Changing the datestyle in Postgres

### DISCUSSION THREAD

**Discussion by Flambe** [supabase/supabase#8855] [Questions] [answered]
I'm using Supabse across the CLI, self-hosted and SaaS.

I am working with legacy data that has some dates stored as strings (E.G. `13.01.2022 09:11:11 Europe/Berlin`). I'm trying to convert them to something more useful via casting: `'13.01.2022 09:11:11 Europe/Berlin'::timestamptz`.

By default, Postgres throws an error telling me that my date is out of range & I need a different datestyle. If I set the datestyle to be correct (`set datestyle to 'ISO, DMY'`) then this problem goes away for the current DB session, but I need a permenant solution. Apparently you can set datestyle via the postgres.conf, but that's not available on SaaS and I can't edit that on CLI as restarting wipes the DB.

Is there a recommended way to set the datestyle in Supabase CLI/Saas/self-hosted?

**Comment by GaryAustin1** [comment, marked as answer]
I waited to answer on this in case someone who really knows had something for you...
Have you tried https://www.postgresql.org/docs/current/sql-alterdatabase.html to change the setting?  
`The remaining forms change the session default for a run-time configuration variable for a PostgreSQL database. Whenever a new session is subsequently started in that database, the specified value becomes the session default value. The database-specific default overrides whatever setting is present in postgresql.conf or has been received from the postgres command line. Only the database owner or a superuser can change the session defaults for a database. Certain variables cannot be set this way, or can only be set by a superuser.`

**Reply by Flambe** [reply]
I've tried using that to update the datestyle, but regardless of if I try to do something like `select '13.01.2022 09:11:11 Europe/Berlin'::timestamptz` immediately afterwards, or if I create a new session and then run that select I still get the same `date/time field value out of range` error.

Perhaps I'm using it wrong since it doesn't seem to apply at all? Though I'm not sure what using it correctly would be in that case.

Thank you for the idea!

**Reply by Flambe** [reply]
I've done some more investigating and it looks like `alter database postgres set datestyle to 'ISO, DMY'` may actually be enough.

I believe my issue was that both of the clients I was testing with (phpstorm & dbeaver) use the jdbc driver to connect to postgres, and this seems to be overriding the datestyle setting on client sessions to be MDY format.

The reason I belive this is because when I check what the source of the setting is from both of those clients (`SELECT setting, source FROM pg_settings WHERE name = 'DateStyle'`), I get back `client` instead of `database`.

Using other clients like supabase studio returns DMY with a source of database like I'd expect.

Not 100% sure that this is completely working yet, but once I am I'll mark this as answered.

**Reply by Flambe** [reply]
Yeah I fully believe it was JDBC causing me issues.

`alter database postgres set datestyle to 'ISO, DMY'` works fine for Supabase, but JDBC ignores it so I have to use `set datestyle to 'ISO, DMY'` whenever I start a new session from phpstorm/dbeaver.
